technical-reports/color/token-naming.md

```diff
@@ -0,0 +1,389 @@
+| -------------------------------------------------------------------------------------------------------------------------------------------------------------------- | --------------------------------------------------------- |
+| Easy mapping between different tokens for color contrast. For example, all 100-400 tokens pair with 500-900 tokens in order to create accessible color combinations. | Less memorable and less obvious difference between tokens |
+| If using bounded scales, the token name can help indicate something about the underlying value                                                                       |                                                           |
+
+### Alias Tokens
+
+For **Alias Tokens**, we recommend grouping tokens with similar intentions by prioritizing the category + property within the name. For example, all background color Alias tokens would likely start with `$color-background-XXX`.
+
+We recommend avoiding abbreviations. For example, use “background” instead of “bg”, to help with clarity.
+
+Here’s how alias tokens may be represented:
+
+<aside class="example">
+
+```scss
+$color-background-error: $color-red-600;
+$color-background-success: $color-green-400;
+$color-text-base: $color-palette-black; // Token with variant
+$color-text-error-hover: $color-red-700; // Token with variant and state
+```
+
+</aside>
+
+Color alias tokens could also be grouped by concept, like so:
+
+<aside class="example">
+
+```scss
+// These can be used for background, border, or text colors
+$color-feedback-error: $color-red-600;
+$color-feedback-success: $color-green-400;
+```
+
+</aside>
+
+### Component Tokens
+
+Component-specific names should start with the component they support and be located close to the component code. They commonly refer to alias tokens and can be helpful when trying to use consistent styles across components while still maintaining separation of concerns.
+
+<aside class="example">
+
+```scss
+$badge-color-background-error: $color-background-error or $color-feedback-error;
+$badge-color-background-success: $color-background-success;
+$badge-color-text-error: $color-text-error or $color-feedback-error;
+```
+
+</aside>
+
+### Supporting color modes
+
+Design tokens can be matched with multiple values to support various color modes, like a light mode and dark mode, as an example. This can be implemented in multiple ways, as demonstrated below.
+
+#### Single declaration
+
+In this case, the **$color-text-default** token is defined once, and “value” is considered the default for light mode, and the “darkValue” is used in dark mode.
+
+<aside class="example">
+
+```json
+// tokens/color/alias.json
+{
+  "color": {
+    "$type": "color",
+    "text": {
+      "default": {
+        "$description": "Default text color",
+        "$value": {
```

Comment:
@adekunleoduye is there any reason for single declarations we can't do:

```
...
"$value": "#111111"
...
```

The additional object with a single key of `"$hex"` seems unnecessary.

Reply:
My understanding was that this notation was still supported?